### PR TITLE
MaterialIDs for solid models preparation

### PR DESCRIPTION
**Pull request by endJunction**
Preparation for [next PR](https://github.com/endJunction/ogs/tree/MaterialIdsForSolidModels).

### CODE REVIEW

File: ProcessLib/LIE/Common/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+APPEND_SOURCE_FILES(SOURCES)
+
+add_library(LIECommon ${SOURCES})
+if(BUILD_SHARED_LIBS)
+    install(TARGETS LIECommon LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
```
[the remark this answers is not in the document]
Comment (bilke) [reply ProcessLib/LIE/Common/CMakeLists.txt]: @chleh Yep, I also thought about automating this... I will come up with a solution..

File: BaseLib/ConfigTree-impl.h [hunk 2]
```diff
@@ -187,6 +187,16 @@ getConfigAttribute(std::string const& attr) const
     error("Did not find XML attribute with name \"" + attr + "\".");
 }
 
+template <typename T>
+T ConfigTree::getConfigAttribute(std::string const& attr,
+                                 T const& default_value) const
```
Comment (TomFischer) [inline BaseLib/ConfigTree-impl.h]: Is the formatting correct?

File: BaseLib/ConfigTree.h [hunk 3]
```diff
@@ -364,18 +364,32 @@ class ConfigTree final
...
     template<typename T> T
     getConfigAttribute(std::string const& attr) const;
 
-    /*! Get XML attribute \c attr of type \c T for the current parameter if present.
+    /*! Get XML attribute \c attr of type \c T for the current parameter or the
+     * \c default_value.
      *
-     * \return the requested attribute
+     * This method has a similar behaviour as getConfigAttribute(std::string
+     * const&) except in case of errors the \c default_value is returned.
```
[the remark this answers is not in the document]
Comment (endJunction) [reply BaseLib/ConfigTree.h]: Improved both.

File: BaseLib/ConfigTree-impl.h (hunk 2 above)
Comment (endJunction) [reply BaseLib/ConfigTree-impl.h]: Yes.

File: BaseLib/ConfigTree.h (hunk 3 above)
[the remark this answers is not in the document]
Comment (chleh) [reply BaseLib/ConfigTree.h]: Thanks.

**Pull request merged**